File: problems/p_7/p7.cpp
/*
project euler problem 7
*/
#include <iostream>

bool isPrime(long num){
  if (num == 1) {
    return false;
  }
  long i = 2;
  while (i * i <= num) {
    if (num % i == 0) {
      return false;
    }
    i++;
  }
  return true;
}

int main(int argc, char const *argv[]) {
  int nextPrime = 3;
  int numberOfPrimes = 2;
  int lastPrime = 0;
  while (numberOfPrimes <= 10001) {
    if (isPrime(nextPrime)) {
      numberOfPrimes += 1;
      lastPrime = nextPrime;
    }
    nextPrime += 2;
  }
  std::cout << lastPrime << '\n';
  // std::cout << isPrime(104743) << '\n';
  return 0;
}

File: problems/p_7/p7.py
"""
project euler problem 7
"""
def is_prime(num):
    if num == 1:
        return False
    i = 2
    while i*i <= num:
        if num % i == 0:
            return False
        i += 1
    return True

next_prime = 3
number_of_primes = 2
last_prime = 0
while number_of_primes <= 10001:
    if is_prime(next_prime):
        number_of_primes += 1
        last_prime = next_prime
    next_prime += 2

print(last_prime)
